devicetypes/iblinds/iblinds-zwave.src/iblinds-zwave.groovy

```diff
@@ -0,0 +1,299 @@
+                attributeState "partially open", label:'Open', action:"close", icon:"st.shades.shade-open", backgroundColor:"#79b821", nextState:"closing"
+                attributeState "opening", label:'${name}', action:"stop", icon:"st.shades.shade-opening", backgroundColor:"#79b821", nextState:"partially open"
+                attributeState "closing", label:'${name}', action:"stop", icon:"st.shades.shade-closing", backgroundColor:"#ffffff", nextState:"partially open"
+            }
+            tileAttribute ("device.level", key: "SLIDER_CONTROL") {
+                attributeState "level", action:"setLevel"
+            }
+        }
+
+        standardTile("home", "device.level", width: 2, height: 2, decoration: "flat") {
+            state "default", label: "home", action:"presetPosition", icon:"st.Home.home2"
+        }
+
+        standardTile("refresh", "device.refresh", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+            state "default", label:'', action:"refresh.refresh", icon:"st.secondary.refresh", nextState: "disabled"
+            state "disabled", label:'', action:"", icon:"st.secondary.refresh"
+        }
+
+        valueTile("battery", "device.battery", decoration: "flat", inactiveLabel: false, width: 2, height: 2) {
+            state "battery", label:'${currentValue}% battery', unit:""
+        }
+
+        preferences {
+            input "preset", "number", title: "Preset position", defaultValue: 50, range: "1..99", required: false, displayDuringSetup: false
+            input "reverse", "bool", title: "Reverse", description: "Reverse Blind Direction",defaultValue: false, required: false , displayDuringSetup: false //input name:, type:
+        }
+
+        main(["windowShade"])
+        details(["windowShade", "home", "refresh", "battery"])
+
+    }
+}
+
+def parse(String description) {
+    def result = null
+    //if (description =~ /command: 2603, payload: ([0-9A-Fa-f]{6})/)
+    // TODO: Workaround manual parsing of v4 multilevel report
+    def cmd = zwave.parse(description, [0x20: 1, 0x26: 3])  // TODO: switch to SwitchMultilevel v4 and use target value
+    if (cmd) {
+        result = zwaveEvent(cmd)
+    }
+    log.debug "Parsed '$description' to ${result.inspect()}"
+    return result
+}
+
+def getCheckInterval() {
+    // iblinds is a battery-powered device, and it's not very critical
+    // to know whether they're online or not – 12 hrs
+    12 * 60 * 60 //12 hours 
+
+}
+
+def installed() {
+    sendEvent(name: "checkInterval", value: checkInterval, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+    sendEvent(name: "supportedWindowShadeCommands", value: JsonOutput.toJson(["open", "close"]), displayed: false)
+    response(refresh())
+}
+
+def updated() {
+    if (device.latestValue("checkInterval") != checkInterval) {
```

Comment:
It may not be needed here (since checkInterval has been set in installed() already).

Reply:
From  published Z-Wave-Window-Shade DTH